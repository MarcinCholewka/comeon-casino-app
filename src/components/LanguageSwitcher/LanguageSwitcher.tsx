import { useMemo } from 'react';
import { useTranslation } from 'react-i18next';
import { GB } from 'country-flag-icons/react/3x2';
import { PL } from 'country-flag-icons/react/3x2';

enum LANGUAGES {
  EN = 'en',
  PL = 'pl',
}

export const LanguageSwitcher = () => {
  const { i18n } = useTranslation();
  const currentLanguage = useMemo(() => i18n.language, [i18n.language]);
  const isCurrentLanguageEN = currentLanguage === LANGUAGES.EN;

  const handleLanguageChange = () =>
    i18n.changeLanguage(isCurrentLanguageEN ? LANGUAGES.PL : LANGUAGES.EN);

  return (
    <div className='absolute z-10 right-1 sm:right-4 top-1 sm-top-4'>
      <label
        htmlFor='lang-switcher'
        className='inline-flex items-center p-2 rounded-md cursor-pointer dark:text-gray-800'>
        <input
          id='lang-switcher'
          type='checkbox'
          className='hidden peer w-full'
          checked={isCurrentLanguageEN}
          onChange={handleLanguageChange}
        />
        <span className='px-4 py-2 rounded-l-md text-white dark:bg-[#8EB50E] peer-checked:dark:bg-stone-200 peer-checked:dark:text-inherit'>
          <PL title='Poland' className='w-[12px] sm:w-[24px]' />
        </span>
        <span className='px-4 py-2 rounded-r-md dark:bg-stone-200 peer-checked:dark:text-white peer-checked:dark:bg-[#8EB50E]'>
          <GB title='England' className='w-[11px] sm:w-[23px]' />
        </span>
      </label>
    </div>
  );
};
