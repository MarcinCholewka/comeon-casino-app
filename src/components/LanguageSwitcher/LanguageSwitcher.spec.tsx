import { render } from '@testing-library/react';
import { LanguageSwitcher } from './LanguageSwitcher';

test('LanguageSwitcher toggles language when clicked', () => {
  const { container, getByTestId } = render(<LanguageSwitcher />);

  // Check initial language
  const languageSwitch = getByTestId('lang-switcher');

  expect(languageSwitch).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
