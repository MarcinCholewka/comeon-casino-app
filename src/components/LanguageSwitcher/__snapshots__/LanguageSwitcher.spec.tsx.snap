// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSwitcher toggles language when clicked 1`] = `
<div>
  <div
    class="absolute z-10 right-1 sm:right-4 top-1 sm-top-4"
  >
    <label
      class="inline-flex items-center p-2 rounded-md cursor-pointer dark:text-gray-800"
      data-testid="lang-switcher"
      for="lang-switcher"
    >
      <input
        checked=""
        class="hidden peer w-full"
        id="lang-switcher"
        type="checkbox"
      />
      <span
        class="px-4 py-2 rounded-l-md text-white dark:bg-[#8EB50E] peer-checked:dark:bg-stone-200 peer-checked:dark:text-inherit"
      >
        <svg
          class="w-[12px] sm:w-[24px]"
          viewBox="0 85.333 512 341.333"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Poland
          </title>
          <g
            fill="#FFF"
          >
            <path
              d="M0 85.337h512v341.326H0z"
            />
            <path
              d="M0 85.337h512V256H0z"
            />
          </g>
          <path
            d="M0 256h512v170.663H0z"
            fill="#D80027"
          />
        </svg>
      </span>
      <span
        class="px-4 py-2 rounded-r-md dark:bg-stone-200 peer-checked:dark:text-white peer-checked:dark:bg-[#8EB50E]"
      >
        <svg
          class="w-[11px] sm:w-[23px]"
          viewBox="0 0 513 342"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            England
          </title>
          <g
            fill="#FFF"
          >
            <path
              d="M0 0h513v341.3H0V0z"
            />
            <path
              d="M311.7 230 513 341.3v-31.5L369.3 230h-57.6zM200.3 111.3 0 0v31.5l143.7 79.8h56.6z"
            />
          </g>
          <path
            d="M393.8 230 513 295.7V230H393.8zm-82.1 0L513 341.3v-31.5L369.3 230h-57.6zm146.9 111.3-147-81.7v81.7h147zM90.3 230 0 280.2V230h90.3zm110 14.2v97.2H25.5l174.8-97.2zm-82.1-132.9L0 45.6v65.7h118.2zm82.1 0L0 0v31.5l143.7 79.8h56.6zM53.4 0l147 81.7V0h-147zm368.3 111.3L513 61.1v50.2h-91.3zm-110-14.2V0h174.9L311.7 97.1z"
            fill="#0052B4"
          />
          <g
            fill="#D80027"
          >
            <path
              d="M288 0h-64v138.7H0v64h224v138.7h64V202.7h224v-64H288V0z"
            />
            <path
              d="M311.7 230 513 341.3v-31.5L369.3 230h-57.6zm-168 0L0 309.9v31.5L200.3 230h-56.6zm56.6-118.7L0 0v31.5l143.7 79.8h56.6zm168 0L513 31.5V0L311.7 111.3h56.6z"
            />
          </g>
        </svg>
      </span>
    </label>
  </div>
</div>
`;
